core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.ts

```diff
@@ -42,12 +44,43 @@ export class DeleteSkillModalComponent extends ConfirmOrCancelModal {
     super(ngbActiveModal);
   }
 
+  fetchTopicIdToDiagnosticTestSkillIds(
+      topicAssignments: AssignedSkill[]): void {
+    let allTopicIds = [];
+    for (let topic of topicAssignments) {
+      allTopicIds.push(topic.topicId);
+    }
+    let topicIdsNotEligibleForUnassignment: string[] = [];
+    this.topicsAndSkillsDashboardBackendApiService
+      .fetchTopicIdToDiagnosticTestSkillIdsAsync(allTopicIds).then(
+        (dict) => {
+          for (let topicId in dict.topicIdToDiagnosticTestSkillIds) {
+            let diagnosticTestSkillIds = (
+              dict.topicIdToDiagnosticTestSkillIds[topicId]);
+            if (diagnosticTestSkillIds.length === 1 &&
+                diagnosticTestSkillIds.indexOf(this.skillId) !== -1) {
+              this.allowDeletionOfSkill = false;
+              topicIdsNotEligibleForUnassignment.push(topicId);
+            }
+          }
+          topicAssignments.filter((topic) => {
+            if (topicIdsNotEligibleForUnassignment.indexOf(
+              topic.topicId) === -1) {
+              return true;
+            }
+            this.topicNamesNotEligibleForUnassignment.push(topic.topicName);
+            return false;
+          });
```

Comment:
I am unsure what is the point of this operation. Why are we filtering and where is the filtered result stored?

Reply:
Refactored, please take a look.